BUILD.md

```diff
@@ -0,0 +1,88 @@
+# Building DLAF
+
+## Build using Spack
```

Comment:
I think this section would benefit with some links to the Spack documentation, so that someone that is completely new to Spack is already pointed in the right direction.

Reply:
Is it ok, with the link at the end of the section?

Reply:
I added a suggestion for a link to the [specs & dependencies](https://spack.readthedocs.io/en/latest/basic_usage.html#specs-dependencies) section of the Spack documentation. Feel free to discard it, but I think this can be solved.